src/main.cpp

```diff
@@ -3,3 +3,127 @@
  */
 
 #include "Arduino.h"
+#include <SimpleFOC.h>
+
+//  Encoder(int encA, int encB , int cpr, int index)
+//  - encA, encB    - encoder A and B pins
+//  - ppr           - impulses per rotation  (cpr=ppr*4)
+//  - index pin     - (optional input)
+Encoder sensor = Encoder(2, 3, 2048); //will we have hardware or software interrupts?
+
+// interrupt routine initialization
+void doA(){sensor.handleA();}
+void doB(){sensor.handleB();}
+//void doIndex(){encoder.handleIndex();} if we have more than 2 hardware interrupts
+
+// BLDC motor & driver instance
+// For Surpass C2216-880KV motor | 14 pole pairs | 0.108 Resistance | 880KV
+BLDCMotor motor = BLDCMotor(14);
+
+//  BLDCDriver3PWM( int phA, int phB, int phC, int enA, int enB, int enC )
+//  - phA, phB, phC - A,B,C phase pwm pins
+//  - enA, enB, enC - enable pin for each phase (optional)
+BLDCDriver3PWM driver = BLDCDriver3PWM(9, 5, 6, 8);
+
+// commander communication instance
+Commander command = Commander(Serial);
+void onMotor(char* cmd){ command.motor(&motor, cmd); }
+
+void setup() {
+
+  // enable/disable quadrature mode
+  sensor.quadrature = Quadrature::ON;
+
+  // check if you need internal pullups, enable if so
+  // sensor.pullup = Pullup::USE_EXTERN;
+
+  // initialise magnetic sensor hardware
+  sensor.init();
+  // link the motor to the sensor
+  motor.linkSensor(&sensor);
+
+  // enable encoder hardware interrupts
+  sensor.enableInterrupts(doA, doB);
+
+  // driver config
+  driver.voltage_power_supply = 12; // Volts
+  // driver.pwm_frequency = 20000; //based on mc used
+  driver.init();
+  // link driver
+  motor.linkDriver(&driver);
+
+  // aligning voltage
+  motor.voltage_sensor_align = 3;
+
+  // choose FOC modulation
+  motor.foc_modulation = FOCModulationType::SinePWM;
+
+  // set torque mode to be used
+  // TorqueControlType::voltage    ( default )
+  // TorqueControlType::dc_current NEEDS CURRENT SENSING
+  // TorqueControlType::foc_current NEEDS CURRENT SENSING
+  // https://docs.simplefoc.com/torque_control#comparison
+  motor.torque_controller = TorqueControlType::voltage;
```

Comment:
Looks like we will have current sensing coming from the MC PCB, is that better? Not sure

Reply:
Current sensing is also an option. It requires a few extra steps but not too much.
Current sensing is supported as such [here](https://docs.simplefoc.com/current_sense#current-sensing-support-per-mcu-architecture) for different MCU architectures.